lower-bounds.project

```diff
@@ -0,0 +1,15 @@
+constraints:
```

Comment:
Why is this file necessary? Won't cabal select the correct bounds for a given version on its own?

Reply:
Yes, cabal will select the correct bounds without specifying this file as the "project file." However, cabal selects the *newest* version of each dependency possible by default. When this file is specified as the "project file" it redefines the constraints to force cabal to choose the *lowest* version of each dependency.

This file is only used in the CI to ensure that `validation` does in fact build when the lowest version of each dependency is selected.

I can remove this from the commit and the"Lower Bounds Check" from the GitHub Actions CI if you would like.

Reply:
That's very cool and useful! Thank you for explaining it to me.